lib/puppet/functions/strftime.rb

```diff
@@ -0,0 +1,14 @@
+# (Documentation in 3.x stub)
+#
+# @since 4.7.0
+#
+Puppet::Functions.create_function(:strftime) do
+  dispatch :format do
+    param 'Variant[Timestamp,Timespan]', :time_object
+    param 'String', :format
```

Comment:
A `String(x)' will give you the default. Didn't see any point in adding it here. Optional arguments have the trade-off of hiding forgotten ones.